core/arch/arm/include/kernel/thread.h

```diff
@@ -497,38 +497,71 @@ void thread_add_mutex(struct mutex *m);
  */
 void thread_rem_mutex(struct mutex *m);
 
+/*
+ * Disables and empties the peralloc RPC cache one reference at a time. If
+ * all threads are idle this function returns true and a cookie of one shm
+ * object which was removed from the cache. When the cache is empty
+ * *is_empty is set to true and the cache is disabled else false. If one
+ * thread isn't idle this function returns false.
+ */
+bool thread_disable_prealloc_rpc_cache(bool *is_empty, uint64_t *cookie);
```

Comment:
Could you get rid of `is_empty` by setting `*cookie` to zero when the cache is empty?